### PR TITLE
bengali asr model with no finetuning 

### PR DESCRIPTION
**Pull request by dzeinali**

### CODE REVIEW

**Comment by sw005320** [comment]
Cool, @dzeinali!
@ftshijt and @DanBerrebbi, please help @dzeinali! to finish this PR.

One major comment from me is that we may carefully set the recipe directory name by considering the other recipe's name, e.g., we can use xx_openslrtyy by following su_openslr36 and jv_openslr35?

**Comment by sw005320** [comment]
ben_openslr53 -> bn_openslr53 since the other recipe names are based on a two-letter code?

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+. utils/parse_options.sh
+
+log "data preparation started"
+
+#mkdir -p ${sbn}
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:31]: Please refer to the other recipes on how to use `db.sh` for database directory

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+workspace=$PWD
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:33]: Normally we assume the user would run the scripts from `egs2/xxx/asr1`, so it won't be necessary for us to add the pwd here

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+data_dir='/ocean/projects/cis210027p/dzeinali/espnet/egs2/sbn/asr1/data'
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:34]: recommend to remove local information

Comment (DanBerrebbi) [inline egs2/bn_openslr53/asr1/local/data.sh:34]: You should put a less specific name for a PR, something like $PWD/data/ this way everybody can use it with not a lot of modification !

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    log "sub-stage 0: Download Data to downloads"
+
+    cd ${data_dir}
+    idxs=("1", "2", "3", "4", "5", "6" "7" "8" "9" "a" "b" "c" "d" "e" "f")
+    for i in "${idxs[@]}"; do
+        wget https://us.openslr.org/resources/53/asr_bengali_${i}.zip
+        unzip -o asr_bengali_${i}.zip
+        rm -f asr_bengali_${i}.zip
+    done
+    mv asr_bengali/* .
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:46]: I don't think `mv` would be efficient, how about directly downloading the data to specific spaces.

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+#mkdir -p ${sbn}
+
+workspace=$PWD
+data_dir='/ocean/projects/cis210027p/dzeinali/espnet/egs2/sbn/asr1/data'
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "sub-stage 0: Download Data to downloads"
+
+    cd ${data_dir}
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:39]: `cd` would be not needed?

**Review by ftshijt — COMMENTED** [review]
Many thanks! There are some comments.

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    idxs=("1", "2", "3", "4", "5", "6" "7" "8" "9" "a" "b" "c" "d" "e" "f")
+    for i in "${idxs[@]}"; do
+        wget https://us.openslr.org/resources/53/asr_bengali_${i}.zip
```
Comment (DanBerrebbi) [inline egs2/bn_openslr53/asr1/local/data.sh:42]: Also, it seems that you downloaded the data into your data folder, we usually do it under a /download folder, you can see here that it is done under the /download directory : https://github.com/espnet/espnet/blob/master/egs2/jv_openslr35/asr1/local/data.sh#L38 (and the variable JAVA points to /downloads through its definition in the file db.sh : https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/asr1/db.sh#L101)

Comment (DanBerrebbi) [reply egs2/bn_openslr53/asr1/local/data.sh:42]: This way you do not have to remove the downloaded files, and they do not intersect with the unzipped data files

File: egs2/TEMPLATE/asr1/db.sh [hunk 7]
```diff
@@ -123,6 +123,7 @@ SPEECHCOMMANDS=downloads
 TOTONAC=downloads
 PRIMEWORDS_CHINESE=downloads
 SEAME=
+SBN=
```
Comment (DanBerrebbi) [inline egs2/TEMPLATE/asr1/db.sh:126]: I think a more specific name could be better, like BENGALI (if not used already), and point it to downloads, as the other variables in that folder
BENGALI=downloads (this way you'll be able to use it in your data.sh file) (see other comment)

File: egs2/bn_openslr53/asr1/conf/decode_asr.yaml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+batch_size: 16
```
Comment (DanBerrebbi) [inline egs2/bn_openslr53/asr1/conf/decode_asr.yaml:1]: maybe just put batch_size: 1 to be aligned with you run.sh file

**Comment by DanBerrebbi** [comment]
Great job @dzeinali , those remarks are just to make your recipe usable more easily by any Espnet user

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+. utils/parse_options.sh
+
+log "data preparation started"
+
+mkdir -p ${BENGALI}
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:31]: It's necessary that you also check the value in advance if it's not defined in `db.sh`

File: egs2/bn_openslr53/asr1/local/data.sh (hunk 4 above)
Comment (ftshijt) [reply egs2/bn_openslr53/asr1/local/data.sh:50]: Is this `mv` necessary?

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+workspace=$PWD
+
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "sub-stage 0: Download Data to downloads"
+
+    cd ${BENGALI}
+    idxs=("1", "2", "3", "4", "5", "6" "7" "8" "9" "a" "b" "c" "d" "e" "f")
+    for i in "${idxs[@]}"; do
+        wget https://us.openslr.org/resources/53/asr_bengali_${i}.zip
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:46]: How about set the target directory for ${BENGALI} directly in wget command by `-O`

Comment (ftshijt) [reply egs2/bn_openslr53/asr1/local/data.sh:46]: In that case, you won't do the `cd` and so on

File: egs2/bn_openslr53/asr1/local/data.sh [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+. utils/parse_options.sh
+
+log "data preparation started"
+if [ -z ${BENGALI+x} ]; then
```
Comment (ftshijt) [inline egs2/bn_openslr53/asr1/local/data.sh:30]: How about following the other recipe's checks: https://github.com/espnet/espnet/blob/master/egs2/commonvoice/asr1/local/data.sh#L27-L31

**Review by ftshijt — COMMENTED** [review]
Thanks! Some minor issues as in comment

**Comment by ftshijt** [comment]
As you may find @dzeinali , there are also some issues in shell scripts, please also check them

**Comment by ftshijt** [comment]
Many thanks for the fixes. The recipe looks cool to me now. I will merge it after you pass all the ci tests.

**Comment by ftshijt** [comment]
Many thanks!

**Pull request merged**